src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreCommand.cs

```diff
@@ -56,7 +56,14 @@ internal sealed class SolutionRestoreCommand
             var menuCommandId = new CommandID(CommandSet, CommandId);
             var menuItem = new OleMenuCommand(
                 OnRestorePackages, null, BeforeQueryStatusForPackageRestore, menuCommandId);
-            commandService?.AddCommand(menuItem);
+
+            // call AddCommand through explicitly moving to UI thread since this is now being 
+            // initiliazed as part of AsynPackage
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
`_vsMonitorSelection.GetCmdUIContextCookie` doesn't access Service so it's fine.

There is already `InitializeAsync` defined which is static and creates this instance.

`NuGetUIThreadHelper` Don't want to use CPS thread helper because of RPS perf which is why used ThreadHelper similar to `RestoreManagerPacakge`